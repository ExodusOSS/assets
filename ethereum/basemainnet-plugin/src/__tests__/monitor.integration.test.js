import { collectBalances, getPrivateSeed, monitorIntegrationTester } from '@exodus/assets-testing'

import assetPlugin from '../index.js'

describe(`basemainnet monitor integration test`, () => {
  monitorIntegrationTester({
    assetPlugin,
    assetName: 'basemainnet',
    seed: getPrivateSeed(),
    availableAssetNames: ['basemainnet'], // this allow reducing testing to just some accets. If not the list is massive
    walletAccountCount: 2,
    safeReportMatcher: {},
    asserter: async ({ wallet, asset }) => {
      // 0xbce047fb7444b66e7d8e370834144bad1191b8bd3ea4aab32f5cd340241ff506
      // 1000000000000000 - 200000000000000 - 4200000000000 ETH l2 fee (implicit) - 19355436461 ETH l1 fee (explicit)
      const results = await collectBalances(wallet)
      expect(results).toEqual({
        exodus_0: {
          basemainnet: {
            balance: '795780644563539 wei',
            spendable: '795780644563539 wei',
            spendableBalance: '795780644563539 wei',
            staked: '0 wei',
            total: '795780644563539 wei',
            unconfirmedReceived: '0 wei',
            unconfirmedSent: '0 wei',
            unstaked: '0 wei',
            unstaking: '0 wei',
          },
        },
        exodus_1: {
          basemainnet: {
            balance: '200000000000000 wei',
            spendable: '200000000000000 wei',
            spendableBalance: '200000000000000 wei',
            staked: '0 wei',
            total: '200000000000000 wei',
            unconfirmedReceived: '0 wei',
            unconfirmedSent: '0 wei',
            unstaked: '0 wei',
            unstaking: '0 wei',
          },
        },
      })
    },
  })
})
