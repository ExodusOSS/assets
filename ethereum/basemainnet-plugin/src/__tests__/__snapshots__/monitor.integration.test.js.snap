exports[`basemainnet monitor integration test should start and stop monitor 1`] = `
{
  "accountStates": {
    "exodus_0": {
      "data": {
        "basemainnet": {
          "t": "object",
          "v": {
            "_version": 1,
            "balance": {
              "t": "numberunit",
              "v": {
                "u": {
                  "BASE": 18,
                  "Gwei": 9,
                  "Kwei": 3,
                  "Mwei": 6,
                  "finney": 15,
                  "szabo": 12,
                  "wei": 0,
                },
                "v": "795780644563539 wei",
              },
            },
            "clarityCursor": "",
            "index": 0,
            "startblock": 0,
            "tokenBalances": {
              "t": "object",
              "v": {},
            },
          },
        },
      },
    },
    "exodus_1": {
      "data": {
        "basemainnet": {
          "t": "object",
          "v": {
            "_version": 1,
            "balance": {
              "t": "numberunit",
              "v": {
                "u": {
                  "BASE": 18,
                  "Gwei": 9,
                  "Kwei": 3,
                  "Mwei": 6,
                  "finney": 15,
                  "szabo": 12,
                  "wei": 0,
                },
                "v": "200000000000000 wei",
              },
            },
            "clarityCursor": "",
            "index": 0,
            "startblock": 0,
            "tokenBalances": {
              "t": "object",
              "v": {},
            },
          },
        },
      },
    },
  },
  "addresses": {
    "exodus_0": {
      "basemainnet": {
        "address": "0x90E481d9A664ebbE4Be180d9501962255463036d",
      },
    },
    "exodus_1": {
      "basemainnet": {
        "address": "0xf6c138C36341138dDFC314a11038dA8264B7Ef09",
      },
    },
  },
  "orders": {
    "data": {},
  },
  "version": {
    "navigator": {
      "name": "nodejs",
      "version": "0.0.1",
    },
    "platform": "Testing",
    "version": "0.0.1",
  },
  "walletAccounts": {
    "data": {
      "exodus_0": {
        "color": "#ff0000",
        "enabled": true,
        "icon": "exodus",
        "index": 0,
        "isMultisig": false,
        "label": "Exodus 0",
        "source": "exodus",
      },
      "exodus_1": {
        "color": "#ff0001",
        "enabled": true,
        "icon": "exodus",
        "index": 1,
        "isMultisig": false,
        "label": "Exodus 1",
        "source": "exodus",
      },
    },
  },
}
`;
